### PR TITLE
introduced link to add bets from dashboard view

### PR DESCRIPTION
**Pull request by maksim-baishev**
Introduced a link to add bets from dashboard view based on the implementation of `RoomView`.

The change wasn't tested at all :D

### CODE REVIEW

File: src/components/Dashboard/Dashboard.tsx [hunk 1]
```diff
@@ -50,83 +58,100 @@ const Dashboard = () => {
...
+                                  ? () => setBetDialog({ game, bet: bet })
+                                  : undefined}
+                          />
+                          {betDialog && (
+                              <BetDialog
+                                  {...betDialog}
+                                  roomId={player.room.id}
+                                  onClose={() => setBetDialog(undefined)}
+                                  onSave={() => {
+                                    getRoom(id as string).then(() => setBetDialog(undefined));
```
Comment (maksim-baishev) [inline src/components/Dashboard/Dashboard.tsx:143]: I'm not sure if this part would work - in RoomView implementation we also call `setRoom` method which seemed to me not necessary

File: src/components/Dashboard/Dashboard.tsx [hunk 2]
```diff
@@ -42,6 +45,11 @@ const useStyles = createUseStyles({
   },
 });
 
+type BetDialogInput = {
```
Comment (maksim-baishev) [inline src/components/Dashboard/Dashboard.tsx:48]: copy pasted the type. If the approach would work we could extract it into a common place

File: src/components/Dashboard/Dashboard.tsx [hunk 3]
```diff
@@ -50,83 +58,100 @@ const Dashboard = () => {
...
+                    ));
+
+                    return (
+                        <TableCell key={player.id} align="center" className={classes.betCell}>
+                          <BetCellContent
+                              bet={bet}
+                              onClick={user.id === player.id
+                                  ? () => setBetDialog({ game, bet: bet })
+                                  : undefined}
+                          />
```
Comment (vetrolamu) [inline src/components/Dashboard/Dashboard.tsx:136]: Thanks a lot for your effort and this feature! And particularly thanks for reusing BetCellContent :).

File: src/components/Dashboard/Dashboard.tsx [hunk 4]
```diff
@@ -50,83 +58,100 @@ const Dashboard = () => {
...
+                          {betDialog && (
+                              <BetDialog
+                                  {...betDialog}
+                                  roomId={player.room.id}
+                                  onClose={() => setBetDialog(undefined)}
+                                  onSave={() => {
+                                    getRoom(id as string).then(() => setBetDialog(undefined));
+                                  }}
+                              />
+                          )}
```
Comment (vetrolamu) [inline src/components/Dashboard/Dashboard.tsx:146]: 1. if my eyes do not deceive me, we will have a lot of dialogs at the same time :). I'd put it on root level, and betDialog would include roomId as well.
2. getting room on save won't help without additional changes, I need to think a bit how is better to handle this place.

File: src/components/Dashboard/Dashboard.tsx (hunk 1 above)
Comment (vetrolamu) [inline src/components/Dashboard/Dashboard.tsx:143]: this id as far as I see you take from url, and it's not applicable for the dashboard. Here we need to use that id which is player.room.id

**Comment by maksim-baishev** [comment]
closing PR as feature is implemented within https://github.com/eurobets/eurobets/pull/47

**Pull request closed without merging**